src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeHelper.cs

```diff
@@ -419,9 +419,14 @@ public static (Size, Rectangle) CalculateTargetLocationAndBounds(Size sourceSize
             int targetHeight = targetRectangle.Height > 0 ? targetRectangle.Height : height;
 
             // Target image width and height can be different to the rectangle width and height.
-            return (new Size(width, height), new Rectangle(targetX, targetY, targetWidth, targetHeight));
+            return (new Size(Sanitize(width), Sanitize(height)), new Rectangle(targetX, targetY, Sanitize(targetWidth), Sanitize(targetHeight)));
         }
 
         private static void ThrowInvalid(string message) => throw new InvalidOperationException(message);
+
+        private static int Sanitize(int input)
```

Comment:
Nitpicking but this could be an array.

Reply:
I'm usually not precious, updated now though